components/Buttons/src/ColorThemer/MDCButtonColorThemer.m

```diff
@@ -21,18 +21,24 @@ @implementation MDCButtonColorThemer
 + (void)applySemanticColorScheme:(nonnull id<MDCColorScheming>)colorScheme
                         toButton:(nonnull MDCButton *)button {
   [button setBackgroundColor:colorScheme.primaryColor forState:UIControlStateNormal];
-  [button setBackgroundColor:colorScheme.primaryColor forState:UIControlStateHighlighted];
-  [button setBackgroundColor:colorScheme.primaryColor forState:UIControlStateSelected];
   [button setBackgroundColor:[colorScheme.onSurfaceColor colorWithAlphaComponent:0.12f]
                     forState:UIControlStateDisabled];
   [button setTitleColor:colorScheme.onPrimaryColor forState:UIControlStateNormal];
-  [button setTitleColor:colorScheme.onPrimaryColor forState:UIControlStateHighlighted];
-  [button setTitleColor:colorScheme.onPrimaryColor forState:UIControlStateSelected];
   [button setTitleColor:[colorScheme.onSurfaceColor colorWithAlphaComponent:0.26f]
                forState:UIControlStateDisabled];
   button.disabledAlpha = 1.f;
 }
 
++ (void)applySemanticColorScheme:(nonnull id<MDCColorScheming>)colorScheme
+                    toFlatButton:(nonnull MDCFlatButton *)flatButton {
+  [flatButton setBackgroundColor:UIColor.clearColor forState:UIControlStateNormal];
+  [flatButton setBackgroundColor:UIColor.clearColor forState:UIControlStateDisabled];
+  [flatButton setTitleColor:colorScheme.primaryColor forState:UIControlStateNormal];
+  [flatButton setTitleColor:[colorScheme.onSurfaceColor colorWithAlphaComponent:0.26f]
+                   forState:UIControlStateDisabled];
+  flatButton.disabledAlpha = 1.f;
```

Comment:
@romoore @yarneo Should we be resetting all of the states to nil before setting our own colors here?

Reply:
I would suggest two parts:

1. Set only those states set by the target class. Nilling-out anything that should fall-back to `.normal`
2. Write unit tests to verify the behavior of all possible states.

Reply:
updated :) PTAL